File: solutions/problem-002/solution.js
var fib = function (n) {
    if (n < 1) return 1;
    return fib(n - 1) + fib(n - 2);
}

var fibWithLimit = function (limit) {
    var sum = 0;
    var currentNumber = 0;
    var fibVal = 0;
    while (true) {
        fibVal = fib(currentNumber);
        if (fibVal >= limit) break;
        if (fibVal % 2 == 0) sum += fibVal;
        currentNumber++;
    }
    return sum;
}

console.log(fibWithLimit(4000000));

File: solutions/problem-002/solution.java
public final class eu_p002_sol {  
  
    public static void main(String[] args) {  
        System.out.println(new eu_p002_sol().run());  
    }  
public String run() {  
        int sum = 0;  
        for (int x = 0; ; x++) {  
            int fib = fibonacci(x);  
            if (fib > 4000000)  
                break;  
            if (fib % 2 == 0)  
                sum += fib;  
        }  
        return Integer.toString(sum);  
    }  
  
    private static int fibonacci(int n) {  
        if (n < 0 || n > 46)  
            throw new IllegalArgumentException();  
        int a = 0;  
        int b = 1;  
        for (int i = 0; i < n; i++) {  
            int c = a + b;  
            a = b;  
            b = c;  
        }  
        return a;  
    }  
} 
